apps/src/blockly/eventHandlers.ts

```diff
@@ -117,3 +120,78 @@ export function reflowToolbox() {
     modalWorkspace?.getFlyout()?.reflow();
   }
 }
+export function updateBlockLimits(event: Abstract) {
+  // This check is to update bubbles that show block limits whenever
+  // blocks on the main workspace are updated.
+  if (
+    event.type !== Blockly.Events.BLOCK_CHANGE &&
+    event.type !== Blockly.Events.BLOCK_MOVE &&
+    event.type !== Blockly.Events.BLOCK_CREATE &&
+    event.type !== Blockly.Events.THEME_CHANGE
+  ) {
+    return;
+  }
+  const blockEvent = event as
+    | BlockChange
+    | BlockMove
+    | BlockCreate
+    | ThemeChange;
+  const blockLimitMap = Blockly.blockLimitMap;
+
+  if (!blockEvent.workspaceId || !blockLimitMap || !(blockLimitMap?.size > 0)) {
+    return;
+  }
+  const eventWorkspace = Blockly.Workspace.getById(
+    blockEvent.workspaceId
+  ) as ExtendedWorkspaceSvg;
+  const allWorkspaceBlocks = eventWorkspace?.getAllBlocks();
+  if (!allWorkspaceBlocks) {
```

Comment:
Will `.getAllBlocks()` always return an array? Do you need to check for this?

Reply:
It will always return an array. In this case the check is really checking whether there's a valid eventWorkspace or not, rather than checking that we have a block array.

Reply:
We could add an explicit workspace check instead, right?
```
if (!eventWorkspace) {
return;
}
```